elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUpdatesTestCase.java

```diff
@@ -552,6 +616,54 @@ private String addKeyStoreWithCredentialReference(String keyStoreName, String st
         }
     }
 
+    private void testAddKeyStoreWithInvalidCredentialReference(String keyStoreName, String store, String alias, String secret) throws Exception {
+        Path resources = Paths.get(KeyStoresTestCase.class.getResource(".").toURI());
+        ModelNode operation = new ModelNode();
+        operation.get(ClientConstants.OPERATION_HEADERS).get("allow-resource-service-restart").set(Boolean.TRUE);
+        operation.get(ClientConstants.OP_ADDR).add("subsystem","elytron").add("key-store", keyStoreName);
+        operation.get(ClientConstants.OP).set(ClientConstants.ADD);
+        operation.get(ElytronDescriptionConstants.PATH).set(resources + "/test.keystore");
+        operation.get(ElytronDescriptionConstants.TYPE).set("JKS");
+        if (store != null) {
+            operation.get(CredentialReference.CREDENTIAL_REFERENCE).get(STORE).set(store);
+        }
+        if (alias != null) {
+            operation.get(CredentialReference.CREDENTIAL_REFERENCE).get(ALIAS).set(alias);
+        }
+        if (secret != null) {
+            operation.get(CredentialReference.CREDENTIAL_REFERENCE).get(CLEAR_TEXT).set(secret);
+        }
+
+        ModelNode response = services.executeOperation(operation);
+        assertEquals(FAILED, response.get(OUTCOME).asString());
+        assertTrue(response.get(FAILURE_DESCRIPTION).asString().contains(CREDENTIAL_REFERENCE));
+    }
+
+    private void testWriteCredentialReferenceAttribute(String resource, String resourceName, String store, String alias, String secret, boolean assertFailed) {
+        ModelNode operation = new ModelNode();
+        operation.get(ClientConstants.OP_ADDR).add("subsystem", "elytron").add(resource, resourceName);
+        operation.get(ClientConstants.OP).set(ClientConstants.WRITE_ATTRIBUTE_OPERATION);
+        operation.get(ClientConstants.NAME).set(CREDENTIAL_REFERENCE);
+        ModelNode credentialReference = new ModelNode();
+        if (store != null) {
+            credentialReference.get(STORE).set(store);
+        }
+        if (alias != null) {
+            credentialReference.get(ALIAS).set(alias);
+        }
+        if (secret != null) {
+            credentialReference.get(CLEAR_TEXT).set(secret);
+        }
+        operation.get(ClientConstants.VALUE).set(credentialReference);
+        ModelNode response = services.executeOperation(operation);
+        if (assertFailed) {
+            assertEquals(FAILED, response.get(OUTCOME).asString());
+            assertTrue(response.get(FAILURE_DESCRIPTION).asString().contains(CREDENTIAL_REFERENCE));
+        } else {
+            assertSuccess(response).get(RESULT);
```

Comment:
Isn't the `.get(RESULT)` part redundant as we don't use that value anywhere anymore?

Reply:
Good catch, this was from copy/pasting from somewhere else. Fixed now.